presto-main/src/main/java/com/facebook/presto/sql/planner/DomainTranslator.java

```diff
@@ -696,22 +697,28 @@ protected ExtractionResult visitInPredicate(InPredicate node, Boolean complement
 
             InListExpression valueList = (InListExpression) node.getValueList();
             checkState(!valueList.getValues().isEmpty(), "InListExpression should never be empty");
-
-            ImmutableList.Builder<Expression> disjuncts = ImmutableList.builder();
-            for (Expression expression : valueList.getValues()) {
-                disjuncts.add(new ComparisonExpression(EQUAL, node.getValue(), expression));
+            Iterator<Expression> expressionIterator = valueList.getValues().iterator();
+            TupleDomain<Symbol> tupleDomain = process(new ComparisonExpression(EQUAL, node.getValue(), expressionIterator.next()), complement)
+                    .getTupleDomain();
+            while (expressionIterator.hasNext() && !tupleDomain.isAll()) {
+                ExtractionResult result = process(new ComparisonExpression(EQUAL, node.getValue(), expressionIterator.next()), complement);
+                if (complement) {
+                    tupleDomain = tupleDomain.intersect(result.getTupleDomain());
```

Comment:
what if `result.remainingExpression` is set? You should either handle such a case or `verify(..)` it does not happen.

Reply:
I think so. This is why both options are "good enough": handling (as in my code snippet) or asserting (with `verify`)